sending/base.py

```diff
@@ -316,9 +314,15 @@ async def unsubscribe_from_topic(self, topic_name: str):
         return await self.parent.unsubscribe_from_topic(topic_name, self.id)
 
     def register_callback(self, fn: Callable, on_predicate: Callable = None):
+        on_predicate = ensure_async(on_predicate) if on_predicate else ensure_async(lambda _: True)
+
+        async def combined_predicates(message: QueuedMessage):
+            topic_matches = message.topic is None or message.topic in self.subscribed_topics
```

Comment:
Might need to update `subscribed_topics` to also check `self.parent.subscribed_topics_by_session[__all_sessions__]` so that you receive messages that are specific to the session but also any global sessions.

We might also want to consider calling it `__not_in_a_session__` so that we have a clear delineation between processing messages in a session vs. callbacks on the singleton manager.

Also you can use `is_subscribed_to_topic` here :eyes: